doc/openapi.yaml

```diff
@@ -0,0 +1,895 @@
+---
```

Comment:
what things were manually changed in this file by you? in which cases do we need to manually update it?

Reply:
this is fully autogenerated

Reply:
what about the schema stuff that you mentioned the other day?

Reply:
ahh yup, the schema has to be modified manually the first time we set $ref if we want to use that to make the file smaller. If not then it's autogenerated